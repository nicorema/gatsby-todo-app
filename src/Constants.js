export const BREAKPOINTS = {
  sm: '640px',
  md: '768px',
  lg: '1024px',
  xl: '1280px',
}

export const BOGOTA_CENTER = {
  lat: 4.624335,
  lng: -74.063644,
}

export const NAVIGATION_LINKS = [
  { link: '/todos-manager', title: 'Todos Manager' },
  { link: '/todos-log', title: 'Todos Log' },
]
