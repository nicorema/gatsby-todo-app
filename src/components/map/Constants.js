export const MAX_TOOLTIP_CHARACTERS = 50
