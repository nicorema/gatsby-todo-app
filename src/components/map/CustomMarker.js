import React from 'react'
import FlagIcon from '../../images/icons/FlagIcon'
import { MAX_TOOLTIP_CHARACTERS } from './Constants'
import { StyledMarker } from './Map.styles'

const CustomMarker = ({ tooltip, lat, lng }) => {
  const renderedToolTip =
    tooltip.length > MAX_TOOLTIP_CHARACTERS
      ? tooltip.slice(0, MAX_TOOLTIP_CHARACTERS) + '...'
      : tooltip

  return (
    <StyledMarker data-testid="customMarker" data-testlocation={{ lat, lng }}>
      <p>{renderedToolTip}</p>
      <FlagIcon />
    </StyledMarker>
  )
}

export default CustomMarker
