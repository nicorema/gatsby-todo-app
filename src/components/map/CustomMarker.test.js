import React from 'react'
import renderer from 'react-test-renderer'
import { MAX_TOOLTIP_CHARACTERS } from './Constants'

import CustomMarker from './CustomMarker'

describe('CustomMarker', () => {
  const testingTooltipText = 'Testing Tooltip'
  const customMarkerCmp = renderer.create(
    <CustomMarker tooltip={testingTooltipText} />
  )

  it('renders correctly', () => {
    const snapShot = customMarkerCmp.toJSON()
    expect(snapShot).toMatchSnapshot()
  })

  it('renders tooltip name when tooltip length less than maximum', () => {
    const tooltip = customMarkerCmp.root.findByType('p')
    expect(tooltip.props.children.length).toEqual(testingTooltipText.length)
  })

  it('renders tooltip trimmed name when tooltip length great than maximum', () => {
    const longTestingtooltip =
      'Lorem ipsum dolor sit amet, consectetuer adipiscing elit. Aenean commodo ligula eget dolor. Aenean m'
    customMarkerCmp.update(<CustomMarker tooltip={longTestingtooltip} />)
    const tooltip = customMarkerCmp.root.findByType('p')
    expect(tooltip.props.children.length).toBe(MAX_TOOLTIP_CHARACTERS + 3)
  })
})
