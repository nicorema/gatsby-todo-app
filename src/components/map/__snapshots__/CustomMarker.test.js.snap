// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CustomMarker renders correctly 1`] = `
<div
  className="sc-gtssRu kBNPkW"
  data-testid="customMarker"
  data-testlocation={
    Object {
      "lat": undefined,
      "lng": undefined,
    }
  }
>
  <p>
    Testing Tooltip
  </p>
  <svg
    height="20pt"
    viewBox="0 0 32 32"
    width="20pt"
    xmlns="http://www.w3.org/2000/svg"
  >
    <g>
      <path
        d="M24.332 16.934l7.535 10.199C26.2 29 13.133 28.602 8.067 26V7.867c5 2.598 18.132 3.067 23.8 1.133zm0 0"
        fill="#FC644F"
      />
      <path
        d="M8.066 25.934l4.067-2-3-9.067-1.067 1.531zm0 0"
        fill="#F1543F"
      />
      <path
        d="M2.535 2.934V21s4.73-1.266 9.531 2.867v-18c-4.8-4.199-9.53-2.933-9.53-2.933zm0 0"
        fill="#FF7058"
      />
      <path
        d="M2 0C1 0 .133.8.133 1.8V32H3.8V1.8C3.8.8 3 0 2 0zm0 0"
        fill="#40596B"
      />
    </g>
  </svg>
</div>
`;
